jupyterhub/tests/test_api.py

```diff
@@ -109,6 +109,50 @@
     assert r.status_code == 200
 
 
+# ----------------
+# Server API tests
+# ----------------
+
+
+def fill_server(model):
+    """Fill a default server model
+
+    Any unspecified fields will be filled with the defaults
+    """
+    user_name = model.get('user', {}).get('name')
+    model.setdefault('name', '')
+    model.setdefault('ready', False)
+    model.setdefault('pending', None)
+    model.setdefault('url', None)
+    model.setdefault(
+        'progress_url', 'PREFIX/hub/api/users/%s/server/progress' % user_name
+    )
+    model.setdefault('started', None)
+    model.setdefault('last_activity', None)
+    model.setdefault('state', None)
+    model.setdefault('user_options', {})
+    return model
+
+
+@mark.server
+async def test_get_servers(app):
+    """
+    Basic test for GET /servers. Run this early for predictable results.
+    """
+    db = app.db
+    r = await api_request(app, 'servers')
+    assert r.status_code == 200
+
+    servers = sorted(r.json(), key=lambda d: d['id'])
+    servers = [normalize_server(server) for server in servers]
+    # Note that there are two servers to start because the `app` fixture
+    # creates a user named "user" and test_referer_check creates an admin user.
+    assert servers == [fill_server(server) for server in servers]
```

Comment:
What exactly does this check? At first I thought it checks that we indeed got two servers back from the API, but it doesn't. Then I thought it was checking that the model/JSON fields returned from the API call were as expected, but now I don't think it does that either.

Reply:
I think at some point while working on this I had it checking two servers individually like how `test_get_users` works but obviously removed that, I'm not sure why now. It should be changed since if we don't get any servers back this will still pass with a false positive. If we remove the `id` field and sort by something like the user name then we could assert two servers in the order we expect.